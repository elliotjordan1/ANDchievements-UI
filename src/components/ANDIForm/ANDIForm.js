import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { onInputChange } from '../../global/helpers';
import { 
  Label, 
  SubmitButton,
  FormInput,
  FormWrapper
} from './styles';

const ANDIForm = ({ name }) => {
  const [imageUrl, setImageUrl] = useState();
  const [role, setRole] = useState();

  const handleSubmit = (event) => {
    event.preventDefault()
    return role;
  };

  return (
  <>
    <FormWrapper onSubmit={handleSubmit}>
      <Label>{name}</Label>
      <div>
        <Label>ANDi Image Url</Label>
        <FormInput placeholder='ANDi image url' value={imageUrl} onChange={onInputChange(setImageUrl)} />
      </div>
      <div>
        <Label>ANDi Project Role</Label>
        <FormInput maxLength={20} placeholder='Role on the project' value={role} onChange={onInputChange(setRole)}/>
      </div>
      <div>
        <SubmitButton type='submit'>SUBMIT</SubmitButton>
      </div>
    </FormWrapper>
  </>
  )};

ANDIForm.propTypes = {
  name: PropTypes.string.isRequired
};

export default ANDIForm;
