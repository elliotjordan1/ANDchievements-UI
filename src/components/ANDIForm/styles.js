import styled from 'styled-components';
import theme from '../../global/theme';

export const Label = styled.div`
    font: 500 16px ${theme.fontFamily.primary};
    letter-spacing: 1px;
    @media ${theme.media.tablet} {
        font-size: 18px;
    };
    @media ${theme.media.desktop} {
        font-size: 20px;
    };
`;

export const FormWrapper = styled.form`
  margin: 16px 24px;
  height: 100%;
`;

export const FormInput = styled.input`
    width: 100%;
    padding: 0 8px;
    box-sizing: border-box;
    border:  1px solid black;
    margin: 8px 0;
    font: 300 14px ${theme.fontFamily.primary};
    height: 40px;
    @media ${theme.media.tablet} {
        font-size: 16px;
        margin: 8px 0 16px 0;
        height: 40px;
        padding-left: 16px;
    };
    @media ${theme.media.desktop} {
        font-size: 18px;
        height: 50px;
        margin: 8px 0 24px 0;
    };
`;

export const SubmitButton = styled.button`
    background-color: ${theme.colours.brand.red};
    width: 120px;
    height: 40px;
    cursor: pointer;
    display: block;
    margin: 8px auto;
    border: none;
    color: white;
    font: 500 14px ${theme.fontFamily.primary};
    letter-spacing: 2px;
    margin-top: 8px;
    :hover{
        opacity: 90%;
    }
    :active{
        background-color: ${theme.colours.brand.red2};
    }
    @media ${theme.media.tablet} {
        width: 160px;
        height: 50px;
        font-size: 16px;
        letter-spacing: 2px;
        margin-top: 16px;
    };
    @media ${theme.media.desktop} {
        width: 200px;
        height: 60px;
        font-size: 18px;
        letter-spacing: 3px;
        margin-top: 24px;
    };
`;