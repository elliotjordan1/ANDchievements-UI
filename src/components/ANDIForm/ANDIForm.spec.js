import React from 'react';
import { render } from '@testing-library/react';
import ANDIForm from './ANDIForm';

describe('Form Modal component', () => {
  it('renders correctly', () => {
    const wrapper = render(<ANDIForm name='Test User' />);
    expect(wrapper).toMatchSnapshot();
  });
});
