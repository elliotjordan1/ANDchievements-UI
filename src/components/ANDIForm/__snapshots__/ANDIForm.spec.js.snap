// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Modal component renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="styles__FormWrapper-p1at4m-1 gPiIqZ"
      >
        <div
          class="styles__Label-p1at4m-0 eKyFat"
        >
          Test User
        </div>
        <div>
          <div
            class="styles__Label-p1at4m-0 eKyFat"
          >
            ANDi Image Url
          </div>
          <input
            class="styles__FormInput-p1at4m-2 gNsONR"
            placeholder="ANDi image url"
            value=""
          />
        </div>
        <div>
          <div
            class="styles__Label-p1at4m-0 eKyFat"
          >
            ANDi Project Role
          </div>
          <input
            class="styles__FormInput-p1at4m-2 gNsONR"
            maxlength="20"
            placeholder="Role on the project"
            value=""
          />
        </div>
        <div>
          <button
            class="styles__SubmitButton-p1at4m-3 keHDhv"
            type="submit"
          >
            SUBMIT
          </button>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="styles__FormWrapper-p1at4m-1 gPiIqZ"
    >
      <div
        class="styles__Label-p1at4m-0 eKyFat"
      >
        Test User
      </div>
      <div>
        <div
          class="styles__Label-p1at4m-0 eKyFat"
        >
          ANDi Image Url
        </div>
        <input
          class="styles__FormInput-p1at4m-2 gNsONR"
          placeholder="ANDi image url"
          value=""
        />
      </div>
      <div>
        <div
          class="styles__Label-p1at4m-0 eKyFat"
        >
          ANDi Project Role
        </div>
        <input
          class="styles__FormInput-p1at4m-2 gNsONR"
          maxlength="20"
          placeholder="Role on the project"
          value=""
        />
      </div>
      <div>
        <button
          class="styles__SubmitButton-p1at4m-3 keHDhv"
          type="submit"
        >
          SUBMIT
        </button>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
